openmc/trackfile.py

```diff
@@ -0,0 +1,183 @@
+        self.identifier = _identifier(dset.name)
+
+        # Construct list of track histories
+        tracks_list = []
+        for particle, start, end in zip(particles, offsets[:-1], offsets[1:]):
+            ptype = ParticleType(particle)
+            tracks_list.append(ParticleTrack(ptype, tracks[start:end]))
+        self.particles = tracks_list
+
+    def __repr__(self):
+        return f'<Track {self.identifier}: {len(self.particles)} particles>'
+
+    def plot(self, axes=None):
+        """Produce a 3D plot of particle tracks
+
+        Parameters
+        ----------
+        axes : matplotlib.axes.Axes, optional
+            Axes for plot
+
+        Returns
+        -------
+        axes : matplotlib.axes.Axes
+            Axes for plot
+
+        """
+        import matplotlib.pyplot as plt
+
+        # Setup axes is one wasn't passed
+        if axes is None:
+            fig = plt.figure()
+            ax = plt.axes(projection='3d')
+            ax.set_xlabel('x [cm]')
+            ax.set_ylabel('y [cm]')
+            ax.set_zlabel('z [cm]')
+        else:
+            ax = axes
+
+        # Plot each particle track
+        for _, states in self.particles:
+            r = states['r']
+            ax.plot3D(r['x'], r['y'], r['z'])
+
+        return ax
+
+    @property
+    def sources(self):
+        sources = []
+        for particle_track in self.particles:
+            particle_type = ParticleType(particle_track.particle)
+            state = particle_track.states[0]
+            sources.append(
+                SourceParticle(
+                    r=state['r'], u=state['u'], E=state['E'],
+                    time=state['time'], wgt=state['wgt'],
+                    particle=particle_type
+                )
+            )
+        return sources
+
+
+class TrackFile(list):
+    """Collection of particle tracks
+
+    This class behaves like a list and can be indexed using the normal subscript
+    notation. Each element in the list is a :class:`openmc.Track` object.
+
+    Parameters
+    ----------
+    filepath : str or pathlib.Path
+        Path of file to load
+
+    """
+
+    def __init__(self, filepath):
+        # Read data from track file
+        with h5py.File(filepath, 'r') as fh:
+            # Check filetype and version
+            check_filetype_version(fh, 'track', _VERSION_TRACK)
+
+            for dset_name in sorted(fh, key=_identifier):
+                dset = fh[dset_name]
+                self.append(Track(dset))
+
+    def plot(self):
+        """Produce a 3D plot of particle tracks
+
+        Returns
+        -------
+        matplotlib.axes.Axes
+            Axes for plot
+
+        """
+        import matplotlib.pyplot as plt
+        fig = plt.figure()
+        ax = plt.axes(projection='3d')
+        ax.set_xlabel('x [cm]')
+        ax.set_ylabel('y [cm]')
+        ax.set_zlabel('z [cm]')
+        for track in self:
+            track.plot(ax)
+        return ax
+
```

Comment:
How about adding some utility convenience features like these. Something that allows a user to easily find tracks that pass through a specified material or cell  

```suggestion


    def filter_by_cell(self, cell: int):
        """Returns tracks that interact within the specified cell ID

        Parameters
        ----------
        cell : the cell ID to use when filtering the interactions
        """

        filtered_tracks = []
        for track in self:
            for particles in track.particles:
                for state in particles.states:
                    if cell == state[6]:
                        filtered_tracks.append(track)
        return filtered_tracks

    def filter_by_material(self, material: int):
        """Returns tracks that interact within the specified material ID

        Parameters
        ----------
        material : the material ID to use when filtering the interactions
        """

        filtered_tracks = []
        for track in self:
            for particles in track.particles:
                for state in particles.states:
                    if material == state[6]:
                        filtered_tracks.append(track)
        return filtered_tracks

```

Reply:
I just pushed an update that includes `filter` methods on both the `Tracks` and `Track` classes that takes a slightly different approach than you have here -- let me know if any of it is unclear.

Reply:
I think any approach is better that the code I put here 😀. Thanks for adding that feature. I shall give it a go before Monday evening. Let me test this new addition but otherwise I think this PR is ready to go.